consensus/consensus_v2.go

```diff
@@ -180,14 +192,31 @@ func (consensus *Consensus) finalizeCommits() {
 		Msg("HOORAY!!!!!!! CONSENSUS REACHED!!!!!!!")
 
 	// Sleep to wait for the full block time
-	consensus.getLogger().Info().Msg("[finalizeCommits] Waiting for Block Time")
+	consensus.getLogger().Info().Msg("[finalCommit] Waiting for Block Time")
 	<-time.After(time.Until(consensus.NextBlockDue))
-
-	// Send signal to Node to propose the new block for consensus
-	consensus.ReadySignal <- struct{}{}
-
 	// Update time due for next block
 	consensus.NextBlockDue = time.Now().Add(consensus.BlockPeriod)
+
+	// If still the leader, send commit sig/bitmap to finish the new block proposal,
+	// else, the block proposal will timeout by itself.
+	if consensus.IsLeader() {
+		if block.IsLastBlockInEpoch() {
+			// No pipelining
+			go func() {
+				consensus.getLogger().Info().Msg("[finalCommit] sending block proposal signal")
+				consensus.ReadySignal <- SyncProposal
+			}()
+		} else {
+			// pipelining
+			go func() {
+				select {
+				case consensus.CommitSigChannel <- commitSigAndBitmap:
+				case <-time.After(6 * time.Second):
```

Comment:
the async block proposal will wait for the commitsig for 4 second and the actual reward logic will wait for 5 seconds, this 6 seconds is just a larger number to make sure the upper level logic don't timeout before the callee logic.

Reply:
Can we move these duration to some const and add some comments there? This is really good explanation and it would  definitely be better to be written in somewhere in code.

Reply:
yes, done.